cmd/start.go

```diff
@@ -93,31 +99,123 @@ func runContainer(cmd *cobra.Command, args []string) {
+		}
+		if testDev == "directory" {
+			testEmptyDir := IsEmpty(Data)
+			if !testEmptyDir {
+				log.Fatal(Data + " is not empty, doing nothing.")
+			}
+			envs = append(envs, "OSD_PATH="+Data)
+			volumeBindings = append(volumeBindings, Data+":"+Data)
+		}
+		if testDev == "blockdev" {
+			meUserName, meID := whoAmI()
+			if meID != "0" {
+				log.Fatal("Hey " + meUserName + "! Run me as 'root' when using a block device.")
+			}
+			// We don't have the logic to do the introspection without using blkid.
+			// Unfortunately, blkid is not available on macOS or Windows
+			if runtime.GOOS == "windows" || runtime.GOOS == "darwin" {
+				log.Fatal("Operating system: " + runtime.GOOS + " is not supported in the scenario")
+			}
+			// We run a couple of test here to ensure the device can be used:
+			// 1. test of the device is accessed by a process (open it with O_EXCL)
+			// 2. test if the device has a partition table and/or a filesystem
+			// 3. test if they are partitions in that partition table (you can have a partition table with 0 partitions)
+
+			// First test: is the device opened by a process?!«
+			testDevOpen, _ := DeviceOpened(Data)
+			if testDevOpen {
+				log.Fatal(Data + " is accessed by another process, doing nothing.")
+			}
+
+			// Second test: search for filesystem and partition table
+			diskFormat := GetDiskFormat(Data)
+			lines := strings.Split(diskFormat, "\n")
+			var fstype, pttype string
+			for _, l := range lines {
+				if len(l) <= 0 {
+					// Ignore empty line.
+					continue
+				}
+				cs := strings.Split(l, "=")
+				if len(cs) != 2 {
+					log.Fatal("blkid returns invalid output: " + diskFormat + ". This potentially mean no partition label on your disk.")
+				}
+				// TYPE is filesystem type, and PTTYPE is partition table type, according
+				// to https://www.kernel.org/pub/linux/utils/util-linux/v2.21/libblkid-docs/.
+				if cs[0] == "TYPE" {
+					fstype = cs[1]
+					log.Fatal(Data + " has a filesystem: " + fstype + ", doing nothing.")
+				} else if cs[0] == "PTTYPE" {
+					// Third test: number of partitions
+					pttype = cs[1]
+					// Now we test if the disk has partition(s)
+					// We know parted will return 2 lines if there is a partition table:
+					//
+					// BYT;
+					// /dev/sdc:100GB:scsi:512:512:gpt:HP LOGICAL VOLUME:;
+					//
+					// So we remove the first 2 lines of the output
+					// The third one is always the partition number
+					partedUselessLinesCount := 2
+					num := GetDiskPartitions(Data)
+					partCount := len(num) - partedUselessLinesCount
+					if partCount != 0 {
+						log.Fatal(Data + " has a partition table type " + pttype + " and " + strconv.Itoa(partCount) + " partition(s) doing nothing.")
+					}
+				}
+			}
+			// If we arrive here, it should be safe to use the device.
+			envs = append(envs, "OSD_DEVICE="+Data)
+			PrivilegedContainer = true
+			volumeBindings = append(volumeBindings, "/dev:/dev")
```

Comment:
Can't we limit the binding on the disk itself rather than exposing the whole /dev ?

Reply:
let's see if we can go with /dev/disk/by instead and resolve the link on /dev/sda for example